BUG: CI on Azure broken with PyTest 7.1

Seems like the new release of PyTest broke the CI on Azure. Ex. https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=16988&view=logs&jobId=7d84eb14-5b07-5555-8e62-838a7038a2b0&j=ede89041-5e8e-5201-ce5c-f68c2cbf02dd&t=141d9d53-a127-58b4-0262-4b9693c8e91f

```
E   pytest.PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
```

https://docs.pytest.org/en/latest/changelog.html

I am not 100% sure it's due to Pytest, but the timing seems to corresponds.

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -172,7 +172,7 @@ stages:
             curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
             python3.8 get-pip.py && \
             pip3 --version && \
-            pip3 install setuptools==59.6.0 wheel numpy==1.18.5 cython==0.29.18 pybind11 pytest pytest-timeout pytest-xdist pytest-env pytest-cov Pillow mpmath pythran==0.10.0 --user && \
+            pip3 install setuptools==59.6.0 wheel numpy==1.18.5 cython==0.29.18 pybind11 pytest==7.0.1 pytest-timeout pytest-xdist pytest-env pytest-cov Pillow mpmath pythran==0.10.0 --user && \
             apt-get -y install gcc-5 g++-5 gfortran-8 wget && \
             cd .. && \
             mkdir openblas && cd openblas && \
@@ -285,7 +285,7 @@ stages:
         Pillow
         pybind11
         pythran==0.10.0
-        pytest
+        pytest==7.0.1
         pytest-cov
         pytest-env
         pytest-timeout

--- a/ci/azure-travis-template.yaml
+++ b/ci/azure-travis-template.yaml
@@ -72,7 +72,7 @@ steps:
     mpmath
     pythran
     pybind11
-    pytest
+    pytest==7.0.1
     pytest-xdist
   displayName: 'Install common python dependencies'
 - ${{ if eq(parameters.test_mode, 'full') }}:

